django/core/cache/backends/locmem.py

```diff
@@ -30,7 +30,9 @@ def __init__(self, name, params):
         BaseCache.__init__(self, params)
         self._cache = _caches.setdefault(name, {})
         self._expire_info = _expire_info.setdefault(name, {})
-        self._lock = _locks.setdefault(name, RWLock())
+        if name not in _locks:
+            _locks[name] = RWLock()
+        self._lock = _locks[name]
```

Comment:
Thanks for noticing this! I wasn't aware of that property of setdefault! I've reverted the above change.